xcube/cli/_gen2/progress_delegate.py

```diff
@@ -0,0 +1,122 @@
+# The MIT License (MIT)
+# Copyright (c) 2020 by the xcube development team and contributors
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of
+# this software and associated documentation files (the "Software"), to deal in
+# the Software without restriction, including without limitation the rights to
+# use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
+# of the Software, and to permit persons to whom the Software is furnished to do
+# so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+from typing import Sequence
+
+import requests
+
+from xcube.cli._gen2.genconfig import CallbackConfig
+from xcube.util.assertions import assert_given
+from xcube.util.progress import ProgressState, ProgressObserver
+
+
+def _format_time(t):
+    """Format seconds into a human readable form.
+    Taken form Dask
+
+    >>> _format_time(10.4)
+    '10.4s'
+    >>> _format_time(1000.4)
+    '16min 40.4s'
+    """
+    m, s = divmod(t, 60)
+    h, m = divmod(m, 60)
+    if h:
+        return "{0:2.0f}hr {1:2.0f}min {2:4.1f}s".format(h, m, s)
+    elif m:
+        return "{0:2.0f}min {1:4.1f}s".format(m, s)
+    else:
+        return "{0:4.1f}s".format(s)
+
+
+class ApiProgressCallbackObserver(ProgressObserver):
```

Comment:
This is not a `ProgressObserver` because none of `on_xxx()` methods have an implementation.